hdfs/src/main/scala/akka/stream/alpakka/hdfs/impl/writer/HdfsWriter.scala

```diff
@@ -53,8 +56,8 @@ private[hdfs] trait HdfsWriter[W, I] {
 @InternalApi
 private[writer] object HdfsWriter {
 
-  def createTargetPath(generator: FilePathGenerator, c: Long): Path =
-    generator(c, System.currentTimeMillis / 1000)
+  def createTargetPath(generator: FilePathGenerator, c: Long, timestamp: Long = -1): Path =
+    generator(c, if (generator.newPathForEachFile) timestamp else currentTimeMillis / 1000)
```

Comment:
I suppose we're making the assumption that the `timestamp` is at the scale we want?  The default behaviour scaled timestamp to seconds, for `newPathForEachFile` we use whatever is provided.